app/helpers/codelist_helper.rb

```diff
@@ -24,7 +34,9 @@ def yaml_to_objects_with_description(entity:, type:, code_displayed_in_name: fal
 
     data = data.collect { |item|
       name = code_displayed_in_name ? "#{item["name"]} (#{item["code"]})" : item["name"]
-      OpenStruct.new(name: name, code: item["code"], description: item["description"])
+      description = t("form.label.#{entity}.#{type}.#{item["code"]}", default: item["description"])
```

Comment:
I'm not sure I understand why we have this translation here. I see it takes the values from `activity.en.yml` form.label.activity.aid_type. Is there an advantage of doing it this way versus what we did before in this method? 
Descriptions tend to be stored on their own files inside vendor (as in `aid_type.yml` inside the vendor folder), not here. That's why I'm a bit confused.

I see you already filter the options on the `aid_type_radio_options` to show only the ones in the `ALLOWED_AID_TYPE_CODES` const, which looks good to me. Having those same options on `activity.en.yml` feels redundant? Also, on `form.label.activity.aid_type` there is one type missing that is in the const (`E01`). Sorry for the brain dump here, I just feel I'm missing the key behind all this so I thought I'd better ask 😅

Reply:
The reason I did it this way was because I thought the ambition was that we left the vendored files alone. See the [discussion here](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/706#discussion_r513534024). If it's OK to alter the descriptions, then yeah, I think changing the description in the vendored files would be a neater solution.

Reply:
That's true. I forgot about the custom descriptions being a change on the vendored file. Sorry about that.
I've also seen that the reason `E01` is not on `activity.en.yml` is because we don't use a custom description for that code. That all makes sense now. Thanks.

In that case I think it's fine as it is since we want to keep the vendor intact. The only thing I would suggest is to correct a couple of missing spaces between words in the custom description of `D01` (there's a `volunteersand`, and a `developingcountries`), and it should be good to go. Thank you @pezholio!